app/views/declaration/procedure_codes.scala.html

```diff
@@ -15,60 +15,94 @@
+        exportsInputText: exportsInputText,
+        tariffDetails: tariffDetails,
+        saveButtons: saveButtons,
+        formHelper: uk.gov.hmrc.play.views.html.helpers.FormWithCSRF)
 
 @(mode: Mode, itemId: String, form: Form[ProcedureCodes], codes: Seq[String])(implicit request: JourneyRequest[_], messages: Messages)
 
 @tariffBodyDesc = { @Html(messages("declaration.type.consignmentTariffText",  components.details_content_link())) }
 
 @tariffBody = { @components.tariff_body_partial(messages("declaration.type.procedureCodeDescListItemOne"), messages("declaration.type.procedureCodeDescListItemTwo"))(tariffBodyDesc)}
 
-    @main_template(
-        title = Title("supplementary.procedureCodes.title", "supplementary.items")
-     ) {
+@additionalCodesTable = {
+    @if(codes.nonEmpty) {
+        @govukTable(Table(
+            rows = codes.zipWithIndex.map{ case(code, index) =>
+                Seq(
+                    TableRow(
+                        content = Text(code),
+                        attributes = Map("id" -> s"additionalProcedureCode-table-row$index-label")
+                    ),
+                    TableRow(
+                        content = HtmlContent(removeButton(Some(code), Some(messages("declaration.procedureCodes.additionalProcedureCode.remove.hint", code)))),
+                        attributes = Map("id" -> s"additionalProcedureCode-table-row$index-remove_button")
+                    )
+                )
+            },
+            head = Some(List(
+                HeadCell(
+                    content = Text(messages("declaration.procedureCodes.additionalProcedureCode.table.header")),
+                    colspan = Some(2)
+                )
+            ))
+        ))
+    }
+}
 
+@govukLayout(
+    title = Title("declaration.procedureCodes.title", "supplementary.items"),
+    backButton = Some(BackButton(messages("site.back"), Navigator.backLink(ProcedureCodes, mode)))) {
 
-        @helper.form(controllers.declaration.routes.ProcedureCodesController.submitProcedureCodes(mode, itemId), 'autoComplete -> "off") {
-            @helper.CSRF.formField
+    @formHelper(action = controllers.declaration.routes.ProcedureCodesController.submitProcedureCodes(mode, itemId), 'autoComplete -> "off") {
 
-            @components.back_link(Navigator.backLink(ProcedureCodes, mode), mode)
+        @errorSummary(form.errors)
 
-            @components.error_summary(form.errors)
+        @sectionHeader(messages("supplementary.items"))
 
-            @components.heading(
-                sectionHeader = messages("supplementary.items"),
-                title = messages("supplementary.procedureCodes.title")
-            )
+        @govukFieldset(Fieldset(
+            legend = Some(Legend(
+                content = Text(messages("declaration.procedureCodes.title")),
+                classes = gdsPageLegend,
+                isPageHeading = true
+            )),
+            html = HtmlFormat.fill(immutable.Seq(
```

Comment:
Nit: `immutable` is redundant here.

Reply:
Doesn't compile without it.  There's 2x types of Seq on this page